axelrod/strategies/handshake.py

```diff
@@ -0,0 +1,33 @@
+from axelrod import Actions, Player, Game
+
+C, D = Actions.C, Actions.D
+
+
+class Handshake(Player):
+    """Starts with C, D. If the opponent plays the same way, cooperate forever,
+    else defect forever."""
+
+    name = 'Handshake'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        Player.__init__(self)
+        self.initial_plays = [C, D]
```

Comment:
There's no drawback, I just added the strategy for completeness with the literature. We can generalize now if you'd like, or leave it as an easy PR for someone else.